### ISSUE TITLE
load-path lost after upgrading doom to recent commit

### ISSUE THREAD

**Issue by felipeAH** [doomemacs/doomemacs#6500]
### What did you expect to happen?

After upgrade to d6d1e600c0b22ce323558002eccdaac6edbcf2b2 commit, doing doom upgrade. **load-path** value is lost when loading **doom-autoloads-file** in **core-start.el** so system installed packages are unavailable.

load-path value not lost

### What actually happened?

load-path value lost so only doom installed packages are available, and system installed packages are not.

### Describe your attempts to resolve the issue

I work around saving and then appending **load-path** after loading  **doom-autoloads-file** in **core-start.el**

;;
;;Save load path
(setq 123/load-path load-path)
(condition-case-unless-debug e
    ;; Avoid `file-name-sans-extension' for premature optimization reasons.
    ;; `string-remove-suffix' is cheaper because it performs no file sanity
    ;; checks; just plain ol' string manipulation.
    (load (string-remove-suffix ".el" doom-autoloads-file) nil 'nomessage)
  (file-missing
   ;; If the autoloads file fails to load then the user forgot to sync, or
   ;; aborted a doom command midway!
   (if (locate-file doom-autoloads-file load-path)
       ;; Something inside the autoloads file is triggering this error;
       ;; forward it to the caller!
       (signal 'doom-autoload-error e)
     (signal 'doom-error
             (list "Doom is in an incomplete state"
                   "run 'doom sync' on the command line to repair it")))))

;; Restore load-path
(setq load-path (append load-path 123/load-path))
```



### Steps to reproduce

upgrade to commit in question, and try to load a system installed package.

### System Information

https://pastebin.com/pkcmFWvF

**Comment by gagbo** [comment]
Hello,

You should manipulate your `load-path` in your `config.el` file if you want to add extra folders. "System installed" packages are still available for me in my `load-path` after running `doom upgrade`, as doing `M-x eval-expression RET (require 'calc) RET` doesn't return any error; and I am exactly on

```
commit d6d1e600c0b22ce323558002eccdaac6edbcf2b2 (HEAD -> master, origin/master, origin/HEAD)
Author: Henrik Lissner <git@henrik.io>
Date:   Tue Jun 21 03:02:27 2022 +0200
```

**Comment by hlissner** [comment]
Duplicate of #6473. Fixed in 3b3c008b1b787bf4755c3286eb4c522c9a1cb071.
